# URL of the first bug: https://sourceforge.net/p/swig/bugs/1379/
# URL of the second bug: https://sourceforge.net/p/swig/bugs/1370/
--- eossdk_wrap.cpp-t
+++ eossdk_wrap.cpp-t
@@ -1618,6 +1618,7 @@
   if (!SwigPyObject_Check(next)) {
     return NULL;
   }
+  ((SwigPyObject *)next)->next = sobj->next;
   sobj->next = next;
   Py_INCREF(next);
   return SWIG_Py_Void();
@@ -3440,7 +3440,11 @@
     }
       
   public:
-    virtual ~SwigPyIterator() {}
+    virtual ~SwigPyIterator() {
+       SWIG_PYTHON_THREAD_BEGIN_BLOCK;
+       _seq = 0;
+       SWIG_PYTHON_THREAD_END_BLOCK;
+    }
 
     // Access iterator method, required by Python
     virtual PyObject *value() const = 0;
